packages/main/src/ColorPalette.ts

```diff
@@ -259,34 +275,80 @@ class ColorPalette extends UI5Element {
 		});
 	}
 
-	_onclick(e: MouseEvent) {
-		const target = e.target as ColorPaletteItem;
-		if (target.hasAttribute("ui5-color-palette-item")) {
-			this.selectColor(target);
+	get effectiveColorItems() {
+		let colorItems = this.colors;
+
+		if (this.popupMode) {
+			colorItems = this.getSlottedNodes<ColorPaletteItem>("colors");
 		}
+
+		return colorItems;
+	}
+
+	/**
+	 * Ensures that only one item is selected or only the last selected item remains active if more than one are explicitly set as 'selected'.
+	 * @private
+	 */
+	_ensureSingleSelectionOrDeselectAll() {
+		const selectedItems = this.allColorsInPalette.filter(item => item.selected);
+		selectedItems.pop();
+		selectedItems.forEach(item => { item.selected = false; });
+	}
+
+	_onclick(e: MouseEvent) {
+		this.handleSelection(e.target as ColorPaletteItem);
 	}
 
 	_onkeyup(e: KeyboardEvent) {
 		const target = e.target as ColorPaletteItem;
-		if (isSpace(e) && target.hasAttribute("ui5-color-palette-item")) {
+		if (isSpace(e)) {
 			e.preventDefault();
-			this.selectColor(target);
+			this.handleSelection(target);
 		}
 	}
 
 	_onkeydown(e: KeyboardEvent) {
 		const target = e.target as ColorPaletteItem;
-		if (isEnter(e) && target.hasAttribute("ui5-color-palette-item")) {
-			this.selectColor(target);
+		if (isEnter(e)) {
+			this.handleSelection(target);
 		}
 	}
 
+	handleSelection(target: ColorPaletteItem) {
+		if (!target.hasAttribute("ui5-color-palette-item") || !target.value) {
+			return;
+		}
+
+		this._shouldFocusRecentColors = false;
+
+		this.selectColor(target);
+
+		// Handle selection for items within the 'recentColorsElements'
+		if (this.recentColorsElements.includes(target)) {
+			this.recentColorsElements[0].selected = true;
+			this.recentColorsElements[0].focus();
+			this._currentlySelected = this.recentColorsElements[0];
+		} else {
+			this.allColorsInPalette.forEach(item => {
+				item.selected = item === target;
+			});
+			this._currentlySelected = target;
+		}
+
+		this._ensureSingleSelectionOrDeselectAll();
+	}
+
 	_onDefaultColorKeyDown(e: KeyboardEvent) {
 		if (isTabNext(e) && this.popupMode) {
 			e.preventDefault();
 			this._onDefaultColorClick();
 		}
 
+		if (isSpace(e) || isEnter(e)) {
```

Comment:
executing some logic that changes the component state from user interaction via space key is not correct. should be on keyup.